src/parser/parser_flow.ml

```diff
@@ -2852,6 +2852,26 @@ end = struct
             extends;
           })) in
 
+      (* This is more or less identical to `type_alias` *)
+      let declare_type_alias env start_loc =
+	Expect.token env T_TYPE;
+	Eat.push_lex_mode env TYPE_LEX;
+	let id = Parse.identifier env in
+	let typeParameters = Type.type_parameter_declaration env in
+	Expect.token env T_ASSIGN;
+	let right = Type._type env in
+	let end_loc = match Peek.semicolon_loc env with
+	  | None -> fst right
+	  | Some end_loc -> end_loc in
+	Eat.semicolon env;
+	Eat.pop_lex_mode env;
+	Loc.btwn start_loc end_loc, Statement.(TypeAlias TypeAlias.({
```

Comment:
This should generate a `DeclareTypeAlias` node instead of a `TypeAlias` node, right?